OpenRA.Mods.Common/SpriteLoaders/IcnD2Loader.cs

```diff
@@ -0,0 +1,274 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Drawing;
+using System.IO;
+using OpenRA;
+using OpenRA.FileFormats;
+using OpenRA.Graphics;
+
+namespace OpenRA.Mods.Common.SpriteLoaders
+{
+	public class IcnD2Loader : ISpriteLoader2
+	{
+		const int SizeX = 16;
+		const int SizeY = 16;
+		const int TileSize = SizeX * SizeY / 2;
+
+		ushort[][] mapTileSet;
+
+		uint ssetOffset, ssetLength;
+		uint rpalOffset, rpalLength;
+		uint rtblOffset, rtblLength;
+		uint numTiles;
+
+		byte[] rtbl;
+		byte[][] rpal;
+
+		class IcnD2Tile : ISpriteFrame
+		{
+			public Size Size { get; private set; }
+			public Size FrameSize { get { return Size; } }
+			public float2 Offset { get { return float2.Zero; } }
+			public byte[] Data { get; set; }
+			public bool DisableExportPadding { get { return false; } }
+
+			public IcnD2Tile(Stream s, byte[] pallette)
+			{
+				byte[] tile = StreamExts.ReadBytes(s, TileSize);
```

Comment:
OpenRA style convention is to use `var` when declaring local variables (unless this introduced ambiguity - the compiler will tell you when that happens).